index.js

```diff
@@ -736,13 +740,14 @@ class Command extends EventEmitter {
    * @param {string[]} [argv]
    * @param {Object} [parseOptions]
    * @param {string} parseOptions.from - where the args are from: 'node', 'user', 'electron'
-   * @return {Promise}
+   * @return {Promise} promise `this`
    * @api public
    */
 
   parseAsync(argv, parseOptions) {
-    this.parse(argv, parseOptions);
-    return Promise.all(this._actionResults).then(() => this);
+    return this._parseProgram(argv, parseOptions).then(() => {
```

Comment:
Instead of scattering `Promise.resolve` through the code in places that do not use Promises, could call `Promise.resolve` here instead to minimise the code churn.